sunpy/net/dataretriever/tests/test_norh.py

```diff
@@ -1,41 +1,141 @@
+                                    NORHClient._get_url_for_timerange(
+                                        map_['TimeRange'], wavelength=wavelength),
+                                    time=time_range)
+        # Attach the client with the QueryResponse
+        resp.client = NORHClient
+        return resp
 
 
 @pytest.mark.remote_data
-@pytest.mark.parametrize("timerange,url_start,url_end", [
-    (TimeRange('2012/4/21', '2012/4/21'),
-     'ftp://anonymous:data@sunpy.org@solar-pub.nao.ac.jp/pub/nsro/norh/data/tcx/2012/04/tca120421',
-     'ftp://anonymous:data@sunpy.org@solar-pub.nao.ac.jp/pub/nsro/norh/data/tcx/2012/04/tca120421'
-     ),
-    (TimeRange('2012/12/1', '2012/12/2'),
-     'ftp://anonymous:data@sunpy.org@solar-pub.nao.ac.jp/pub/nsro/norh/data/tcx/2012/12/tca121201',
-     'ftp://anonymous:data@sunpy.org@solar-pub.nao.ac.jp/pub/nsro/norh/data/tcx/2012/12/tca121202'
-     ),
-    (TimeRange('2012/3/7', '2012/3/14'),
-     'ftp://anonymous:data@sunpy.org@solar-pub.nao.ac.jp/pub/nsro/norh/data/tcx/2012/03/tca120307',
-     'ftp://anonymous:data@sunpy.org@solar-pub.nao.ac.jp/pub/nsro/norh/data/tcx/2012/03/tca120314'
-     )
-])
-def test_get_url_for_time_range(timerange, url_start, url_end):
-    urls = norh.NoRHClient()._get_url_for_timerange(timerange, wavelength=17*u.GHz)
+def test_fetch_working():
+    """
+    Tests if the mock fetch contains the correct data.
+    """
+
+    qr1 = NORHClient.search(a.Time('2012/10/4', '2012/10/6'),
+                            a.Instrument('norh'), a.Wavelength(17*u.GHz))
+
+    # Create a mock query object
+    mock_qr = mock_query_object('2012/10/4', '2012/10/6', wavelength=17*u.GHz)
+
+    mock_qr = mock_qr[0]
+    qr = qr1[0]
+    # Assert the values
+    assert mock_qr.source == qr.source
+    assert mock_qr.instrument == qr.instrument
+    assert mock_qr.physobs == qr.physobs
+    assert mock_qr.provider == qr.provider
+    assert mock_qr.url == qr.url
+    assert mock_qr.time == qr.time
+
+    # Assert if the time range is same
+    assert qr1.time_range() == TimeRange('2012/10/4', '2012/10/7')
+
+    # Assert the fetch object, and whether it returns the correct set of files
+    res = NORHClient.fetch(qr1)
+    download_list = res.wait(progress=False)
+    assert len(download_list) == len(qr1)
+
+    download_list.sort()
+    # Assert each of the files have the same file names
+    assert download_list[0].split('/')[-1].split('.')[0] == 'tca121004'
+    assert download_list[1].split('/')[-1].split('.')[0] == 'tca121005'
+    assert download_list[2].split('/')[-1].split('.')[0] == 'tca121006'
+
+@pytest.fixture
+def create_mock_url(mocker,sdate,edate,wave):
+    mocker.patch('sunpy.net.dataretriever.sources.norh.NoRHClient._get_url_for_timerange',
+                return_value=create_url(sdate,edate,wave))
+
+@pytest.fixture
+def create_mock_fetch(mocker, sdate, edate, wave):
+    mocker.patch('sunpy.net.fido_factory.Fido.fetch', side_effect=(
```

Comment:
This isn't the correct type of object to return from Fido.fetch. Also you shoudn't mock `Fido.fetch` you should mock the calls inside `parfive` to ensure that the download is being enqued correctly. See `test_noaa` for the kind of thing I mean: https://github.com/sunpy/sunpy/blob/master/sunpy/net/dataretriever/tests/test_noaa.py#L110